roles/download/tasks/download_container.yml

```diff
@@ -1,19 +1,26 @@
 ---
-- name: container_download | Make download decision if pull is required by tag or sha256
-  include_tasks: set_docker_image_facts.yml
-  when:
-    - download.enabled
-    - download.container
-  tags:
-    - facts
-
 - block:
+    - name: set default values for flag variables
+      set_fact:
+        image_is_cached: false
+        image_changed: false
+        pull_required: "{{ download_always_pull }}"
+      tags:
+        - facts
+
     - name: download_container | Set a few facts
       import_tasks: set_container_facts.yml
-      run_once: "{{ download_run_once }}"
       tags:
         - facts
 
+    - name: download_container | Prepare container download
+      include_tasks: check_pull_required.yml
+      when:
+        - not download_always_pull
+
+    - debug:
+        msg: "XXX Pull {{ image_reponame }} required is: {{ pull_required }}"
```

Comment:
Why the extra string "XXX"? Can you remove it?

Reply:
no problem
This line has been here a long time